guides/assets.md

```diff
@@ -0,0 +1,240 @@
+
+A typical assets directory for a Solidus application will look like:
+
+    app
+    |-- assets
+        |-- images
+        |   |-- spree
+        |       |-- frontend
+        |       |-- backend
+        |-- javascripts
+        |   |-- spree
+        |       |-- frontend
+        |       |   |-- all.js
+        |       |-- backend
+        |           |-- all.js
+        |-- stylesheets
+        |   |-- spree
+        |       |-- frontend
+        |       |   |-- all.css
+        |       |-- backend
+        |           |-- all.css
+
+Solidus also generates four top level manifests (all.css & all.js, see
+above) that require all the core extension's and site specific
+stylesheets / JavaScript files.
+
+## How core extensions (engines) manage assets
+
+All core engines have been updated to provide four asset manifests that
+are responsible for bundling up all the JavaScript files and stylesheets
+required for that engine.
+
+For example, Solidus provides the following manifests:
+
+    vendor
+    |-- assets
+        |-- javascripts
+        |   |-- spree
+        |       |-- frontend
+        |       |   |-- all.js
+        |       |-- backend
+        |           |-- all.js
+        |-- stylesheets
+        |   |-- spree
+        |       |-- frontend
+        |       |   |-- all.css
+        |       |-- backend
+        |           |-- all.css
+
+These manifests are included by default by the
+relevant all.css or all.js in the host Solidus application. For example,
+`vendor/assets/javascripts/spree/backend/all.js` includes:
+
+```ruby
+    //= require jquery
+    //= require jquery_ujs
+
+    //= require spree/backend
+
+    //= require_tree .
+```
+
+External JavaScript libraries, stylesheets and images have also be
+relocated into vendor/assets (again Rails 3.1 standard approach), and
+all core extensions no longer have public directories.
+
+## Managing your application's assets
+
+Assets that customize your Solidus store should go inside the appropriate
+directories inside `vendor/assets/images/spree`, `vendor/assets/javascripts/spree`,
+or `vendor/assets/stylesheets/spree`. This is done so that these assets do
+not interfere with other parts of your application.
+
+## Managing your extension's assets
+
+We're suggesting that all third party extensions should adopt the same
+approach as Solidus and provide the same four (or less depending on
+what the extension requires) manifest files, using the same directory
+structure as outlined above.
+
+Third party extension manifest files will not be automatically included
+in the relevant all.(js|css) files so it's important to document the
+manual inclusion in your extensions installation instructions or provide
+a Rails generator to do so.
+
+For an example of an extension using a generator to install assets and
+migrations take a look at the [install_generator for spree_fancy](https://github.com/spree/spree_fancy/blob/master/lib/generators/spree_fancy/install/install_generator.rb).
```

Comment:
Why don't we use https://github.com/solidusio/solidus_auth_devise/blob/master/lib/generators/solidus/auth/install/install_generator.rb